nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenFTP.java

```diff
@@ -0,0 +1,215 @@
+    private final AtomicReference<ProcessSessionFactory> sessionFactory = new AtomicReference<>();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void startFtpServer(ProcessContext context) {
+        if (ftpServer == null) {
+            String username = context.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
+            String password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
+            String bindAddress = context.getProperty(BIND_ADDRESS).evaluateAttributeExpressions().getValue();
+            int port = context.getProperty(PORT).evaluateAttributeExpressions().asInteger();
+
+            try {
+                sessionFactorySetSignal = new CountDownLatch(1);
+                ftpServer = new NifiFtpServer.Builder()
+                        .sessionFactory(sessionFactory)
+                        .sessionFactorySetSignal(sessionFactorySetSignal)
+                        .bindAddress(bindAddress)
+                        .port(port)
+                        .username(username)
+                        .password(password)
+                        .build();
+                ftpServer.start();
+            } catch (ProcessException processException) {
+                getLogger().error(processException.getMessage(), processException);
+                stopFtpServer();
+                throw processException;
+            }
+        } else {
+            getLogger().warn("Ftp server already started.");
+        }
+    }
+
+    @OnStopped
+    public void stopFtpServer() {
+        if (ftpServer != null && !ftpServer.isStopped()) {
+            ftpServer.stop();
+        }
+        ftpServer = null;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFactory) throws ProcessException {
+        if (this.sessionFactory.compareAndSet(null, sessionFactory)) {
+            sessionFactorySetSignal.countDown();
+        }
+        context.yield();
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext context) {
+        List<ValidationResult> results = new ArrayList<>(2);
+        String username = context.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
+        String password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
+
+        if ((username == null) && (password != null)) {
+            results.add(usernameOrPasswordIsNull(USERNAME, PASSWORD));
+        } else if ((username != null) && (password == null)) {
+            results.add(usernameOrPasswordIsNull(PASSWORD, USERNAME));
+        } else if ((username != null) && (password != null)) {
+            validateAgainstEmptyString(username, USERNAME, results);
```

Comment:
We must make sure this check is performed even if the NON_BLANK_VALIDATOR is removed from the PropertyDescriptor for some reason. At this point I would rather remove the NON_BLANK_VALIDATOR than break the completeness of this check.

Reply:
I think this is something we do not have to prepare. Duplication because of possible future changes can lead into strange corners. I would think, no one will remove a validator without a strong reason and reviews